test/Microsoft.DotNet.Cli.Utils.Tests/DangerousFileDetectorTests.cs

```diff
@@ -0,0 +1,109 @@
+    {
+        [WindowsOnlyFact]
+        public void ItShouldDetectFileWithMarkOfTheWeb()
+        {
+            var testFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+            File.WriteAllText(testFile, string.Empty);
+            AlternateStream.WriteAlternateStream(
+                testFile,
+                "Zone.Identifier",
+                "[ZoneTransfer]\r\nZoneId=3\r\nReferrerUrl=C:\\Users\\test.zip\r\n");
+
+            new DangerousFileDetector().IsDangerous(testFile).Should().BeTrue();
+        }
+
+        [Fact]
+        public void WhenThereIsNoFileItReturnsFalse()
+        {
+            var testFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+
+            new DangerousFileDetector().IsDangerous(testFile).Should().BeFalse();
+        }
+
+        [NonWindowsOnlyFact]
+        public void WhenRunOnNonWindowsReturnFalse()
+        {
+            var testFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+            File.WriteAllText(testFile, string.Empty);
+
+            new DangerousFileDetector().IsDangerous(testFile).Should().BeFalse();
+        }
+
+        private static class AlternateStream
+        {
+            private const uint GenericWrite = 0x40000000;
+
+            public static void WriteAlternateStream(string filePath, string altStreamName, string content)
+            {
+                if (string.IsNullOrWhiteSpace(filePath))
+                {
+                    throw new ArgumentException("message", nameof(filePath));
+                }
+
+                if (string.IsNullOrWhiteSpace(altStreamName))
+                {
+                    throw new ArgumentException("message", nameof(altStreamName));
+                }
+
+                if (content == null)
+                {
+                    throw new ArgumentNullException(nameof(content));
+                }
+
+                string altStream = filePath + ":" + altStreamName;
+
+                SafeFileHandle fileHandle
+                    = CreateFile(
+                        filename: altStream,
+                        desiredAccess: GenericWrite,
+                        shareMode: 0,
+                        attributes: IntPtr.Zero,
+                        creationDisposition: (uint)FileMode.CreateNew,
+                        flagsAndAttributes: 0,
+                        templateFile: IntPtr.Zero);
+
+                if (!fileHandle.IsInvalid)
+                {
+                    using (var streamWriter = new StreamWriter(new FileStream(fileHandle, FileAccess.Write)))
+                    {
+                        streamWriter.WriteLine(content);
+                        streamWriter.Flush();
+                    }
+                }
+                else
+                {
+                    Exception ex = new Win32Exception(Marshal.GetLastWin32Error());
+                    if (!ex.Message.Contains("cannot find the file"))
+                    {
+                        throw ex;
```

Comment:
I know this is just for test, but `throw` is better than `throw ex` because it preserves the original stack trace.

Reply:
this is a marshaled exception, we need to throw here. Or it will just be silent 

`Exception ex = new Win32Exception(Marshal.GetLastWin32Error());`

Reply:
there no recatch, to get the I need to `throw ex` to get the exception at all